criu/sockets.c

```diff
@@ -631,8 +647,13 @@ int dump_socket_opts(int sk, SkOptsEntry *soe)
 	ret |= dump_opt(sk, SOL_SOCKET, SO_PRIORITY, &soe->so_priority);
 	soe->has_so_rcvlowat = true;
 	ret |= dump_opt(sk, SOL_SOCKET, SO_RCVLOWAT, &soe->so_rcvlowat);
-	soe->has_so_mark = true;
+	/*
+	 * Restoring SO_MARK requires root or CAP_NET_ADMIN. Avoid saving it
+	 * in unprivileged mode if still has its default value.
+	 */
 	ret |= dump_opt(sk, SOL_SOCKET, SO_MARK, &soe->so_mark);
+	if (soe->so_mark != 0)
+		soe->has_so_mark = true;
```

Comment:
nit: this if does not look nice to me =)

1) Normally we don't write `if (var != 0)` in criu, instead we use shorter `if (var)`.
2) But in this exact case the whole construction can be simplified, the form:
```
if (!var)
        has_var = true;
```
can be replaced with
```
has_var = !!var;
```
As far as I see all has_* variables are of type protobuf_c_boolean which is int basically, so it is better to use "!!" to convert non-zero value of var to 1.

Reply:
Both ways work for me. @Snorch thanks for review.